### ISSUE TITLE
install fails on non-windows when ALIEN_INSTALL_TYPE=share 

### ISSUE THREAD

**Issue by shawnlaffan** [Perl5-Alien/Alien-MSYS#9]
The alienfile has conditions to use a system install for non-windows, but these seem to be overridden by ALIEN_INSTALL_TYPE.

```sh
ALIEN_INSTALL_TYPE=share cpanm Alien::MSYS
```

Possibly the alienfile could have something like this near the top or in the probe sub (untested):
```perl
delete local $ENV{ALIEN_INSTALL_TYPE}
  if $^O ne 'MSWin32';
```

If you think it's a reasonable approach, and have a preferred location, then I'll work up a PR.  

Regards,
Shawn.

**Comment by shawnlaffan** [comment]
Sorry, I neglected to note that the install fails with that variable set and when the OS is not windows.  Hopefully it could be inferred well enough.

**Comment by plicease** [comment]
Localizing the `delete` won't work because the probe actually happens after the `alienfile` is evaluated.  In general I'd prefer not to override the `ALIEN_INSTALL_TYPE` environment variable, but I think in this case it probably is reasonable.  A share install never makes sense in Unix.  I think a non-localized `delete` is a little icky but probably acceptable in this case.

**Comment by plicease** [comment]
It might make sense to allow a share install on `cygwin`.  I think at least one point this did work, but I no longer test on `cygwin`, tbh.  If you do make a PR, please allow `cygwin` to do a share install as well.

**Comment by shawnlaffan** [comment]
Yes, it is a bit icky to be messing with environment variables.  

I won't localise it and will cover both ```cygwin``` and ```MSWin32```.

**Comment by plicease** [comment]
This should be fixed in 0.14.  Thanks for the report and the PR.
